app/scripts/controllers/threebox.js

```diff
@@ -0,0 +1,181 @@
+const ObservableStore = require('obs-store')
+const Box = process.env.IN_TEST
+  ? require('../../../development/mock-3box')
+  : require('3box/dist/3box.min')
+// const Box = require(process.env.IN_TEST ? '../lib/mock-3box' : '3box/dist/3box.min')
+const log = require('loglevel')
+
+const JsonRpcEngine = require('json-rpc-engine')
+const providerFromEngine = require('eth-json-rpc-middleware/providerFromEngine')
+const createMetamaskMiddleware = require('./network/createMetamaskMiddleware')
+const createOriginMiddleware = require('../lib/createOriginMiddleware')
+
+class ThreeBoxController {
+  constructor (opts = {}) {
+    const {
+      preferencesController,
+      keyringController,
+      addressBookController,
+      version,
+      getKeyringControllerState,
+      getSelectedAddress,
+      signPersonalMessage,
+    } = opts
+
+    this.preferencesController = preferencesController
+    this.addressBookController = addressBookController
+    this.keyringController = keyringController
+    this.provider = this._createProvider({
+      static: {
```

Comment:
What is this property for? It looks as though it's being ignored in `createMetamaskMiddleware`

Reply:
Yep, it can be removed